src/backport.ts

```diff
@@ -86,19 +90,48 @@ export class Backport {
 
       let commitShasToCherryPick: string[];
 
-      // find out if "squashed and merged" or "rebased and merged"
-      if (await this.github.isSquashed(mainpr) || mainpr.commits == 1) {
-        console.log("PR was squashed and merged");
-        // if squashed, then use the merge_commit_sha
-        commitShasToCherryPick = [
-          await this.github.getMergeCommitSha(mainpr),
-        ]?.filter(Boolean) as string[];
-      } else {
-        // if rebased, then use all the commits from the original PR
-        console.log("PR was rebased and merged");
-        commitShasToCherryPick = commitShas;
+      /*
+
+      squashed:
+        - > single parent
+        - > the parent of the `merge_commit_sha` is NOT associated with the PR
+
+        => use merge_commit_sha
+
+      rebased single-commit:
+        - > single parent
+        - > the parent of the `merge_commit_sha` is associated with the PR
+
+        => use commits associated with the PR
+
+      rebased multi-commit:
+        - > single parent,
+        - > the parent of the `merge_commit_sha` is associated with the PR
+
+        => use commits associated with the PR
+
+      merge-commit:
+        - > multiple parents
+
+        => use commits associated with the PR
+      */
+      // switch case to check if it is a squash, rebase, or merge commit
+      switch (await this.github.mergeStrategy(mainpr)) {
+        case MergeStrategy.SQUASHED:
+          commitShasToCherryPick = [
+            await this.github.getMergeCommitSha(mainpr),
+          ]?.filter(Boolean) as string[];
+        case MergeStrategy.REBASED:
+          commitShasToCherryPick = commitShas;
+        case MergeStrategy.MERGECOMMIT:
+          commitShasToCherryPick = commitShas;
+        case MergeStrategy.UNKNOWN:
+          // probably write a comment
+          console.log("Could not detect merge strategy.");
+          return;
+        default:
+          commitShasToCherryPick = [];
```

Comment:
😍 I really like the idea using a switch over an enum here

🔧 On the UNKNOWN case let's just cherrypick the commits from the pull request. Doing something is likely better for this action than doing nothing. But we should log a warning or so.

❓ What does the `[..]?filter(Boolean) as string[]` do?

Reply:
> 🔧 On the UNKNOWN case let's just cherrypick the commits from the pull request. Doing something is likely better for this action than doing nothing. But we should log a warning or so.

👍🏼 

> What does the [..]?filter(Boolean) as string[] do?

This function accesses the `merge_commit_sha` attribute which, _may_ (no source) be null. This was just me being defensive and filtering out the non string values. I'm not a javascript person, so feel free to suggest 🙇🏼

Reply:
>I'm not a javascript person, so feel free to suggest

Me neither 😄 I used this project to explore typescript while solving a problem our team was facing. I've improved over the years but I'm still not fully accustomed to it. Let's leave it for now, and focus on the core changes. We can always change it later.